Algorithm/QCAlgorithm.History.cs

```diff
@@ -647,18 +648,71 @@ private HistoryRequest CreateHistoryRequest(Security security, SubscriptionDataC
             return request;
         }
 
-        private static SubscriptionDataConfig GetMatchingSubscription(Security security, Type type)
+        private SubscriptionDataConfig GetMatchingSubscription(Symbol symbol, Type type)
         {
             // find the first subscription matching the requested type with a higher resolution than requested
-            return GetMatchingSubscriptions(security, type).FirstOrDefault();
+            return GetMatchingSubscriptions(symbol, type).FirstOrDefault();
+        }
+
+        private IEnumerable<SubscriptionDataConfig> GetMatchingSubscriptions(Symbol symbol, Type type)
+        {
+            Security security;
+            if (Securities.TryGetValue(symbol, out security))
+            {
+                // find all subscriptions matching the requested type with a higher resolution than requested
+                return from sub in security.Subscriptions.OrderByDescending(s => s.Resolution)
+                       where type.IsAssignableFrom(sub.Type)
+                       select sub;
+            }
+            else
+            {
+                var resolution = UniverseSettings.Resolution;
+                var timeZone = GetExchange(symbol).TimeZone;
+                var subscriptionDataTypes = SubscriptionManager.LookupSubscriptionConfigDataTypes(symbol.SecurityType, resolution, symbol.IsCanonical());
+                return subscriptionDataTypes.Select(x => new SubscriptionDataConfig(x.Item1, symbol, resolution, timeZone, timeZone, UniverseSettings.FillForward, UniverseSettings.ExtendedMarketHours, false, false, x.Item2));
+            }
+        }
+
+        private SecurityExchange GetExchange(Symbol symbol)
+        {
+            Security security;
+            if (Securities.TryGetValue(symbol, out security))
+            {
+                return security.Exchange;
+            }
+
+            var exchangeHours = MarketHoursDatabase.GetEntry(symbol.ID.Market, symbol, symbol.ID.SecurityType).ExchangeHours;
+
+            switch (symbol.SecurityType)
+            {
+                case SecurityType.Equity:
+                    return new Securities.Equity.EquityExchange(exchangeHours);
+                case SecurityType.Option:
+                    return new Securities.Option.OptionExchange(exchangeHours);
+                case SecurityType.Forex:
+                    return new Securities.Forex.ForexExchange(exchangeHours);
+                case SecurityType.Future:
+                    return new Securities.Future.FutureExchange(exchangeHours);
+                case SecurityType.Cfd:
+                    return new Securities.Cfd.CfdExchange(exchangeHours);
+                case SecurityType.Crypto:
+                    return new Securities.Crypto.CryptoExchange(exchangeHours);
```

Comment:
do we actually _need_ the exchange object? I think we just need the hours and the time zone which are all contained within the `SecurityExchangeHours`